docs/theming.md

```diff
@@ -0,0 +1,147 @@
+
+```python
+{% block footer %}
+    <div id="footer">
+        <div class="container">
+          <p>{{site_title}}</p>
+          <p>Site created using <a href="https://github.com/open-oni/open-oni">open-oni</a> software, built off the Library of Congress's <a href="https://github.com/LibraryofCongress/chronam">chronam</a>.</p>
+        </div>
+    </div>
+{% endblock footer %}
+```
+
+That block "footer" code is very convenient, because it means that we can define our own block in the overrides and we don't have to change anything in core, nor copy the entire `__base.html` file into our theme!
+
+Open up `themes/<your-theme-name>/templates/__overrides.html`.  Now you can copy in the `{% block footer %} {% endblock footer %}` code and put anything you like in it.
+
+```python
+{% block footer %}
+  Put your own HTML code in here!!!
+{% endblock footer %}
+```
+
+Let's say that you want to add some JavaScript to a page.  Here's where things get a little different.  You probably don't want to override the JS that is already operating on each page, so your first instinct might be to copy the existing block and add your code at the bottom.  But wait!  There's a neat method that will help you!  You can use `{{ block.super }}` to tell the app "use all the original code plus my new code!!"  Here's how it looks.
+
+**The original javascript block in __base.html**
+
+```python
+{% block javascript %}
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
+    <script type="text/javascript" async="" src="{% static 'js/highlight.js' %}"></script>
+    <script type="text/javascript" async="" src="{% static 'vendor/bootstrap/js/bootstrap.min.js' %}"></script>
+{% endblock javascript %}
+```
+
+**Your override for the javascript block in __overrides.html**
+
+```python
+{% block javascript %}
+    {{ block.super }}
+    <script>    // your code here!    </script>
+{% endblock javascript %}
+```
+
+## Common Overrides
+
+As described above, you can override blocks to use your own HTML.  These blocks may vary by page, for example, `/search/pages/results` has many more blocks than `/about`, so you may need to consult the `core/templates` file to learn more about overriding specific parts.  However, the overall structure of the base template has the following blocks which may be overridden in `themes/<your-theme-name>/templates/__overrides.html`:
+
+**`<head>` Content**
+Meta tags, title, CSS inclusion, etc
+
+```
+head_all
+  head_page_meta
+  head_page_title
+  head_page_css
+  head_page_top_js
+  head_item_metadata
+  head_opensearch
+  head_extra
+```
```

Comment:
This list of blocks to override etc has been updated since this theming documentation was created: https://github.com/open-oni/open-oni/blob/feature/docs/themes/default/templates/__overrides.html

Perhaps it would be wise to update the docs here and remove the longer comments from the file I linked to?

Reply:
Sounds like a good idea. That's been the trend so far everywhere except the solr documentation (although I'd be down for moving that, too), so I will work on moving the theming stuff into here.

Reply:
I'm late to the party, but I think it's wise to keep comments in the code that actually gets used.  If you override docs, or copy the settings example, or whatever, you *have* to look at the file because... you're looking at the file.  Not everybody will look at the docs, though.

As an example, any docs I produce for the admin commands will have to exist in the commands themselves first and foremost, and then be mirrored into a `docs/` page.  CLI users can then see the docs without worrying about opening a browser, going to the right tagged release page's `docs/` folder, etc.